src/aiperf/records/records_manager.py

```diff
@@ -59,6 +63,26 @@
 from aiperf.records.phase_completion import PhaseCompletionChecker
 
 
+@dataclass
+class TelemetryTrackingState:
+    """
+    Tracks telemetry-related state and performance metrics.
+
+    Consolidates error tracking, warnings, endpoint status, and performance
+    statistics for GPU telemetry collection and processing.
+    """
+
+    error_counts: dict[ErrorDetails, int] = field(default_factory=dict)
```

Comment:
you can also do this as a defaultdict(int) to use += 1
```suggestion
    error_counts: dict[ErrorDetails, int] = field(default_factory=defaultdict(int))
```